src-tauri/src/aux_functions/adj_matrix.rs

```diff
@@ -10,21 +10,24 @@ use std::collections::HashMap;
 /// # Returns
 ///
 /// * `Vec<Vec<f64>>` - an adjacency matrix
-pub fn convert_to_adj_matrix(graph: &Graph) -> Vec<Vec<f64>> {
-    let mut adj_matrix: Vec<Vec<f64>> = Vec::new();
+pub fn convert_to_adj_matrix(graph: &Graph) -> (Vec<Vec<f64>>, HashMap<usize, String>) {
```

Comment:
I'm hesitant to return a tuple in Rust since it can be somewhat arbitrary as to the meaning of the result. I'd probably lean towards making this a new `struct` type

Reply:
This will also be handled by global state. However, even then, I was thinking of returning tuple because this function won't be exposed to anything other than the global state struct, which will know how to handle the results of this function.

Reply:
Also, that's the point of tuples right? They are ordered so you know which element is what.

Reply:
Yeah that is the point, but they provide no information on how the tuple is meant to be consumed. Seeing `t[0]` is way less informative then `t.matrix`, especially if that tuple grows to contain multiple fields of the same type

Reply:
Well in this instance I think it's more efficient since the size will not grow and the elements are of different types. The function signature is clear enough that you'll know t[0] is adj matrix because adj matrix cannot be represented via HashMap<usize, String>. But again, nobody really needs to know this because state will have fields adj_matrix and etc.